crates/apps_lib/src/cli.rs

```diff
@@ -7918,6 +7944,48 @@ pub mod args {
         }
     }
 
+    #[derive(Clone, Debug)]
+    pub struct SignOffline {
+        pub tx_path: PathBuf,
+        pub secret_keys: Vec<common::SecretKey>,
+        pub owner: Address,
```

Comment:
I think these can be wallet aliases. key/address look-ups are still done offline
```suggestion
        pub secret_keys: Vec<WalletKeypair>,
        pub owner: WalletAddress,
```

Reply:
i think `client utils` cli methods dont have access to the wallet

Reply:
you have access to `args::Global` in `sign_offline` (from `crates/apps_lib/src/client/utils.rs`). this struct holds a `base_dir`, and a `is_pre_genesis` arg. you can either load a pre-genesis or post-genesis wallet with this.

Reply:
for an example you can see `ClientUtils::EpochSleep` cli handle that constructs a ctx

Reply:
Building a context requires a client which requires a connection to a node, I want this to be completly offline

Reply:
```rs
    pub struct Global {
        pub is_pre_genesis: bool,
        pub chain_id: Option<ChainId>,
        pub base_dir: PathBuf,
        pub wasm_dir: Option<PathBuf>,
    }
```
you have access to a value of this type. using `Global`, you can load the wallet by navigating to `$base_dir/$chain_id/wallet.toml` or `$base_dir/pre-genesis/wallet.toml` (depending on whether we're post or pre genesis, respectively). this can be done fully offline. for pre-genesis stuff, we have some examples in the genesis flow code, that is used to sign genesis txs.

Reply:
solve with some help from tiago!